diff -up ppl-1.1/demos/ppl_lcdd/ppl_lcdd.cc.orig ppl-1.1/demos/ppl_lcdd/ppl_lcdd.cc
--- ppl-1.1/demos/ppl_lcdd/ppl_lcdd.cc.orig	2016-03-17 17:48:24.795619726 -0300
+++ ppl-1.1/demos/ppl_lcdd/ppl_lcdd.cc	2016-03-17 17:49:59.364623348 -0300
@@ -224,7 +224,7 @@ std::istream* input_stream_p = 0;
 
 void
 set_input(const char* file_name) {
-  if (input_stream_p && *input_stream_p != std::cin)
+  if (input_stream_p && (void*)input_stream_p != (void*)&std::cin)
     delete input_stream_p;
 
   if (file_name) {
@@ -250,7 +250,7 @@ std::ostream* output_stream_p = 0;
 
 void
 set_output(const char* file_name) {
-  if (output_stream_p && *output_stream_p != std::cout)
+  if (output_stream_p && (void*)output_stream_p != (void*)&std::cout)
     delete output_stream_p;
 
   if (file_name) {
diff -up ppl-1.1/demos/ppl_pips/ppl_pips.cc.orig ppl-1.1/demos/ppl_pips/ppl_pips.cc
--- ppl-1.1/demos/ppl_pips/ppl_pips.cc.orig	2016-03-18 12:01:23.174424242 -0300
+++ ppl-1.1/demos/ppl_pips/ppl_pips.cc	2016-03-18 12:02:23.743426561 -0300
@@ -538,7 +538,7 @@ std::istream* input_stream_p = 0;
 
 void
 set_input(const char* file_name) {
-  if (input_stream_p && *input_stream_p != std::cin)
+  if (input_stream_p && (void*)input_stream_p != (void*)&std::cin)
     delete input_stream_p;
 
   if (file_name) {
@@ -558,7 +558,7 @@ std::ostream* output_stream_p = 0;
 
 void
 set_output(const char* file_name) {
-  if (output_stream_p && *output_stream_p != std::cout)
+  if (output_stream_p && (void*)output_stream_p != (void*)&std::cout)
     delete output_stream_p;
 
   if (file_name) {
