diff -ur ppl-0.9/configure ppl-0.9-patched/configure
--- ppl-0.9/configure	2006-03-12 09:11:32.000000000 +0100
+++ ppl-0.9-patched/configure	2008-09-08 15:38:34.000000000 +0200
@@ -14929,7 +14929,7 @@
   swi_prolog_version=`$swi_prolog -dump-runtime-variables | grep PLVERSION= | sed 's/PLVERSION="\(.*\)";/\1/'`
 
 
-  SWI_PROLOG_INCLUDE_OPTIONS="-I${swi_prolog_base}/include"
+  SWI_PROLOG_INCLUDE_OPTIONS="-I${swi_prolog_base}/include -I/usr/include/pl"
   ac_save_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$CPPFLAGS $SWI_PROLOG_INCLUDE_OPTIONS"
   ac_ext=cc
